Add course cards to home page

noisecapella opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Fixes #240 

#### What's this PR do?
Adds four course cards to the home page using data from the course frontmatter

#### How should this be manually tested?
- Use an updated version of ocw-to-hugo to create markdown for the site. It needs to include https://github.com/mitodl/ocw-to-hugo/pull/101 and https://github.com/mitodl/ocw-to-hugo/pull/102
 - View the homepage. You should see up to 4 course cards sorted from most recently published


#### Screenshots
See comment below

pdpinch commented:
Thanks @noisecapella Can you add mobile screenshots?

aliceriot commented on site/themes/multi_course/layouts/home.html:
```diff
@@ -4,11 +4,45 @@
+  <div class="new-courses d-flex flex-column align-items-center">
+    {{ $pages := .Site.Pages.ByPublishDate.Reverse }}
+    {{ $coursePages := where $pages "Type" "==" "course" }}
+    {{ $childPages := where $coursePages ".Parent.Type" "==" "courseindex" }}
```
I think here you may also be able to check if `layout == "course_home"` - I was just thinking that may be more durable in the long term since the `courseindex` page is more or less a placeholder for use during development that I think we're going to get rid of at some point.

@gumaerc do you have any thoughts about that?

noisecapella replied:
Is `layout` a field on the parent?

aliceriot replied:
it's a field which is set on each page. we have a layout (`course_home`) which is only used for the course home pages

aliceriot replied:
(would be good for @gumaerc to confirm that's true!)

noisecapella replied:
Looks like it works. The count for `.Parent.Type == courseindex` matches `.Params.layout == course_home`

abdulkdawson replied:
@aliceriot I think we may have to put a limit on the list of instructors and topics so that all the cards remain the same length.

noisecapella replied:
@abdulkdawson Is it fine pick the first two instructors and first two topics in the list? I don't know if the instructors are in any particular order

gumaerc replied:
Sorry I'm late on this, and yea, `course_home` is only used for the course home page and is a valid way to verify that the page is being rendered with the `course_home.html` layout template.

aliceriot reviewed (changes requested):
looks good, just a few questions

aliceriot commented:
one other question, probably for @abdulkdawson, should the cards have the same height regardless of their contents?

noisecapella commented:
@pdpinch here are a couple. Looks like I need padding under the card to separate from the other cards. Also, what do you think about the number of cards per row for each width? 

wide:
![Screenshot_2020-10-19 Hugo Course Publisher](https://user-images.githubusercontent.com/863262/96496242-fab9e780-1216-11eb-8aef-238f39e509de.png)

narrow:
![Screenshot_2020-10-19 Hugo Course Publisher(1)](https://user-images.githubusercontent.com/863262/96496257-fee60500-1216-11eb-9d5c-510e9c5ed8fb.png)

aliceriot commented:
one more thing I just noticed, I think we probably need some commas to make the separations between the different topics more clear:

![Screenshot from 2020-10-19 14-47-26](https://user-images.githubusercontent.com/6207644/96498509-24c0d900-121a-11eb-9e1d-7b3fe9d516c5.png)

noisecapella commented on src/css/home.scss:
```diff
@@ -136,3 +136,46 @@ $spacer8: map-get($spacers, 8);
+
+      .card-line {
+        font-size: 12px;
+        padding-top: 10px;
```
Sure, I'll do some investigating

noisecapella commented:
New screenshots:
![Screenshot from 2020-10-19 16-24-15](https://user-images.githubusercontent.com/863262/96507807-b256f580-1227-11eb-9e78-3056e5e6407a.png)

![Screenshot from 2020-10-19 16-24-35](https://user-images.githubusercontent.com/863262/96507815-b5ea7c80-1227-11eb-94ba-43b1bcef11e9.png)

![Screenshot from 2020-10-19 16-24-56](https://user-images.githubusercontent.com/863262/96507822-b84cd680-1227-11eb-8f5e-75df0a1d05e1.png)

noisecapella commented:
I changed it to have the card use the full height necessary for the card, and to trim topics and instructors to 3 each. Updated screenshots:

desktop:
![Screenshot from 2020-10-20 15-13-15](https://user-images.githubusercontent.com/863262/96633429-17bdeb80-12e7-11eb-9388-8965380e54a3.png)

tablet:
![Screenshot from 2020-10-20 15-13-38](https://user-images.githubusercontent.com/863262/96633444-1db3cc80-12e7-11eb-904b-58c09234129a.png)

mobile:
![Screenshot from 2020-10-20 15-13-53](https://user-images.githubusercontent.com/863262/96633457-20aebd00-12e7-11eb-9cca-749a6ce78b9c.png)

abdulkdawson commented:
@noisecapella Why are the cards so wide in mobile?

abdulkdawson commented:
@noisecapella can you provide me with the width and height of the cards in the mobile view?

noisecapella commented:
@abdulkdawson They are a little over 500px. What should the widths be for the cards on different breakpoints?

abdulkdawson commented:
@noisecapella the width for the mobile breakdown should be 373px, Tablet should be 350px and desktop is 300px.

noisecapella commented:
@abdulkdawson It looks like the max width of the container on desktop is 1140px so 300px is too wide for 4 course cards. Should I widen the max container width?

abdulkdawson commented:
@noisecapella Yes, you should defiantly widen the max container width.

abdulkdawson commented:
@noisecapella The proper width for the container should be 1440px.

noisecapella commented:
New screenshots:
![Screenshot_2020-10-21 Hugo Course Publisher(4)](https://user-images.githubusercontent.com/863262/96743409-f960fa00-1391-11eb-80cc-01b0147942a5.png)

![Screenshot_2020-10-21 Hugo Course Publisher(5)](https://user-images.githubusercontent.com/863262/96743420-fc5bea80-1391-11eb-88a0-b6ca93bbe05b.png)

![Screenshot_2020-10-21 Hugo Course Publisher(6)](https://user-images.githubusercontent.com/863262/96743428-febe4480-1391-11eb-9138-45cf216a4c04.png)

aliceriot reviewed (approved):
LGTM! :+1:

abdulkdawson commented:
@noisecapella I think I have communicated the wrong specs to you. I meant the page width should be 1440px I've included a screenshot of the cards with the end to end measurement. 
<img width="681" alt="Screen Shot 2020-10-21 at 3 00 27 PM" src="https://user-images.githubusercontent.com/49685474/96770351-536fb880-13ae-11eb-8e8d-d782e7099879.png">
